website/content/api-docs/system/namespaces.mdx

```diff
@@ -94,3 +94,98 @@ $ curl \
   "path": "ns1/"
 }
 ```
+
+## Lock Namespace
+
+This endpoint locks the api for the current namespace path or optional subpath.
+
+| Method | Path                    |
+| :----- | :---------------------- |
+| `POST`  | `/sys/namespaces/api-lock/lock/:subpath` |
+
+### Sample Request - Current Namespace
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    http://127.0.0.1:8200/v1/sys/namespaces/api-lock/lock
+```
+
+### Sample Response - Current Namespace
+
+```json
+{
+    "unlock_key": "<unlock key for current/ns/path>"
+}
+```
+
+### Sample Request - Descendant of Current Namespace
+
+```shell-session
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    http://127.0.0.1:8200/v1/sys/namespaces/api-lock/lock/some/descendant/subpath
+```
+
+### Sample Response - Descendant of Current Namespace
+
+```json
+{
+    "unlock_key": "<unlock key for current/ns/path/some/descendant/subpath>"
+}
+```
+
+## Unlock Namespace
+
+This endpoint unlocks the api for the current namespace path or optional subpath.
+
+| Method | Path                    |
+| :----- | :---------------------- |
+| `POST`  | `/sys/namespaces/api-lock/unlock/:subpath` |
+
+### Sample Payload - Current Namespace Non-Root
+
+```json
+{
+  "unlock_key": "<unlock key for current/ns/path>"
+}
+```
+
+### Sample Request - Current Namespace Non-Root
```

Comment:
It might be useful to provide some examples that use the `X-Vault-Namespace` header for comparison.

Reply:
0b5ae79b2